core/src/main/java/ch/cyberduck/core/DefaultOAuthPrefixService.java

```diff
@@ -0,0 +1,80 @@
+package ch.cyberduck.core;
+
+/*
+ * Copyright (c) 2002-2023 iterate GmbH. All rights reserved.
+ * https://cyberduck.io/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Default OAuth Prefix Service.
+ * Will just return fields from passed bookmark.
+ */
+public class DefaultOAuthPrefixService implements OAuthPrefixService {
+    private final Host bookmark;
+    private final Protocol protocol;
+
+    public DefaultOAuthPrefixService(Host bookmark) {
+        this.bookmark = bookmark;
+        this.protocol = bookmark.getProtocol();
+    }
+
+    @Override
+    public String getDescription() {
+        if (StringUtils.isNotBlank(bookmark.getCredentials().getUsername())) {
```

Comment:
@AliveDevil we expose `getUsername()` in the same interface - have `getDescription()` return `bookmark.getProtocol().getDescription();` in the default implementation and  extract processing to `DefaultHostPasswordStore`
```
private String getOAuthPrefix(OAuthPrefixService oAuthPrefix) {
        if (StringUtils.isNotBlank(oAuthPrefix.getUsername())) {
            return String.format("%s (%s)", oAuthPrefix.getDescription(),
                    oAuthPrefix.getUsername());
        }
        return oAuthPrefix.getDescription()
}
```
?

Reply:
Move `static getOAuthPrefix` back to DefaultHostPasswordStore then.